### PR TITLE
Update Script #8

### PR DESCRIPTION
**Pull request by z0ne323**

### CODE REVIEW

File: script_vts.py [hunk 1]
```diff
@@ -207,78 +183,61 @@ def formating_file():
...
+            response_file_reputation = file_reputation_api_request(api_key, string_argument)
+            report_data_dic.append("[+] Response from File reputation request: \n")
+            report_data_dic.append(response_file_reputation)
+    return report_data_dic
 
 
 if __name__ == '__main__':
-
-    if checking_arguments(sys.argv) is None:  # Check if an argument is provided
+    if verify_at_least_one_command_line_argument(sys.argv) is None:  # Check if an argument is provided
```
Comment (nigleweber) [inline script_vts.py:224]: Excellent code segment.

File: script_vts.py [hunk 2]
```diff
@@ -18,184 +18,160 @@ def exception_handler(print_exception=False, exception=""):
...
+        prompt = checking_path(prompt)
     except ValueError as val_err:
         exception_handler(True, val_err)
-        return
-
+        return None
     return prompt
 
 
+def checking_path(raw_path):
```
Comment (nigleweber) [inline script_vts.py:50]: Great function. Let's rename it to something more generic such as: `validate_path` or simply `check_if_filepath_exists`.

Additionally let's try to strip out the references to the API file specifically. That way you can use this function for any future programs.

File: script_vts.py [hunk 3]
```diff
@@ -18,184 +18,160 @@ def exception_handler(print_exception=False, exception=""):
...
-def checking_arguments(arguments):
+def verify_at_least_one_command_line_argument(arguments):
     """Function checking the number of arguments"""
     if len(arguments) > 1:
         return arguments
-    return
+    return None
+
+
+def parse_arguments():
```
Comment (nigleweber) [inline script_vts.py:31]: Noice!
Can you see the difference this makes for code readability? Especially in your main function.

File: script_vts.py [hunk 4]
```diff
@@ -207,78 +183,61 @@ def formating_file():
...
     file_arg = False  # Set file argument to false until we know there's one
 
     if args.file:  # Set to true if we find that it is used as cmdline argument, Remove any blank line so we can send proper string to the api_req()
         file_arg = True
-
         formating_file()
 
     string = ""
 
     path = get_path(string)  # Call the get_path function to have the path of the API Key File
```
Comment (nigleweber) [inline script_vts.py:239]: Remove this line, the above and below lines are related.

**Comment by nigleweber** [comment]
Great work on this PR. 👍 

I'll add some more comments to the issue later. For now, the one big item is to start identifying the sections of your code. Let's keep the section name generic for now since we're gonna need to refactor section 2 and 3.
Section1: `Main and setup` - Everything that happens before you make the call to VT
Section2: `Data Handling` - Everything we do with the data up until it's ready to be written to file
Section3: `Data formatting + file write` - Once we've parsed the data we need, we can start formatting strings etc. and finally write to the file.
Section4: `Helpers` - These are functions we've made generic enough that any program could call them. We'll move them into `Helpers.py` soon.

Use something similar to this for denoting sections:
```
//##########################################################
//######## MAIN AND SETUP
//##########################################################
```

File: script_vts.py (hunk 3 above)
Comment (z0ne323) [reply script_vts.py:31]: Yes I totally see the difference thanks for the recommendation clearly the code is way better!

File: script_vts.py (hunk 4 above)
Comment (z0ne323) [reply script_vts.py:239]: Hum, weird this line is actually removed there's no space between `file_arg = True` and `formating_file`. Is it good like this or do you want me to put a space between the two instructions?

**Pull request merged**